cdi-core/src/main/java/com/linkedin/cdi/extractor/MultistageExtractor.java

```diff
@@ -987,6 +993,10 @@ protected void readPayloads(State state) {
     }
   }
 
+  private boolean shouldReadAsBinary(JsonObject entryObject) {
+    return entryObject.has("format") && entryObject.get("format").getAsString().equals("binary");
```

Comment:
Check JsonUtils for a relevant helper method. 
It will take care of null checks and everything.

Reply:
sure will check